lab2.js

```diff
@@ -74,42 +99,87 @@ assert(blob.hoursToOoze(1000, 1) === hoursSpentInDowington,
 // PROBLEM 2: Universal Translator. 20 points
 //*********************************************************
 
-var hello = {
-  klingon: 'nuqneH',  // home planet is Qo'noS
-  romulan: 'Jolan\'tru', // home planet is Romulus
-  'federation standard': 'hello' // home planet is Earth
-};
+//var hello = {
+  //klingon: 'nuqneH',  // home planet is Qo'noS
+  //romulan: 'Jolan\'tru', // home planet is Romulus
+  //'federation standard': 'hello' // home planet is Earth
 
+function SentientBeing(homePlanet, language) {
+  this.homePlanet = homePlanet;
+  this.language = language;
+}
+SentientBeing.prototype.sayHello = function(sb) {
+  var helloReturn = sb.language;
+  switch (helloReturn) {
+    case 'Klingon':
+      return ('nuqneH');
+    case 'Romulan':
+      return ('Jolan\'tru');
+    default:
+      return ('Hello');
+  }
+  var helloLog = this.language;
+  switch (helloLog) {
+    case 'Klingon':
+      console.log('nuqneH');
+      break;
+    case 'Romulan':
+      console.log('Jolan\'tru');
+      break;
+    default:
+      console.log('Hello');
```

Comment:
can also do
console.log(hello[this.language]);
return hello[sb.language];
if you have your instances set up correctly. This is a great use case for bracket notation.